c-dependencies/js-compute-runtime/builtins/request-response.cpp

```diff
@@ -2384,8 +2384,162 @@ bool Response::redirect(JSContext *cx, unsigned argc, JS::Value *vp) {
   return true;
 }
 
+namespace {
+static bool callbackCalled;
+static bool write_json_to_buf(const char16_t *str, uint32_t strlen, void *out) {
+  callbackCalled = true;
+  auto outstr = static_cast<std::u16string *>(out);
+  outstr->reserve(1 + outstr->length() + strlen);
+  outstr->append(str, strlen);
```

Comment:
I think you don't need the `1+` in the reserve here. There's no `0` written to the end, and the call to `shrink_to_fit` on line 2419 is going to remove it through a reallocation anyway. At that point, you can drop the call to `reserve` entirely, as that will happen automatically in the `append`.